### PR TITLE
Feat/dsm: team requests [jira:MATD-50]

### PR DESCRIPTION
**Pull request by karthik01455**
- API endpoints for upsert, get, and delete is created for team requests.
- Also included filters on team requests based on type,status,startdate,enddate, author and search key word through query params.
- Swagger documentation for all end points.
- Validation using joi and middleware.

GET(including filters) :
image

image

POST:
image
image

PUT:
image

DELETE:
image

### CODE REVIEW

File: src/services/dsm/teamRequests.services.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            author,
+            content,
+            status,
+            type,
+            createdAt,
+            ...(taggedIndividuals && { taggedIndividuals }),
+        },
+        ...selectOnlyValidTeamrequestsFields
+    });
+    console.log(createdRequest);
```
Comment (Abhishek-Kharpal) [inline src/services/dsm/teamRequests.services.js:27]: We don't need console log here

File: src/services/dsm/teamRequests.services.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        ...selectOnlyValidTeamrequestsFields
+    }
+    );
+    return teamRequests;
+}
+// service to edit team requests
+const editTeamRequest = async (requestId, author, content, status, type, createdAt, taggedIndividuals) => {
+    const updatedRequest = await prisma.Request.update({
+        where: {
+            requestId: parseInt(requestId),
```
Comment (Abhishek-Kharpal) [inline src/services/dsm/teamRequests.services.js:77]: Id is already parsed through middleware

File: src/controllers/dsm/teamRequests.controller.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+}
+// controller for deleting team request
+const deleteTeamRequest = async (req, res, next) => {
+    try{
+        const { requestId } = req.params;
+        const deletedRequest = await teamRequestsServices.deleteTeamRequest(requestId);
+        res.status(204).json(deletedRequest);
+    }
+    catch(error)
+    { console.log(error);
```
Comment (Abhishek-Kharpal) [inline src/controllers/dsm/teamRequests.controller.js:66]: We don't need to log the error here.

File: src/routes/api/dsm/teamRequests.routes.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+ *         description: team request deleted successfully
+ *       400:
+ *         description: Bad request if not acceptable requestId is passed
+ *       404:
+ *         description: Not found if no team request found with irequestIdd
+ *       500:
+ *         description: Internal server error
+*/
+router.route('/:requestId')
+    .put(parseIntIdParam,generateValidationMiddleware(requestSchema.editTeamRequest), editTeamRequest)
```
Comment (Abhishek-Kharpal) [inline src/routes/api/dsm/teamRequests.routes.js:235]: Are we validating the requestId here?

File: src/services/dsm/teamRequests.services.js [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    endDate,
+    searchKeyword,
+    status,
+    page,
+    limit) => {
+    let filterObj = {};
+    // using query params for filter requests with startDate, endDate, search keyword, status, page , limit and author) 
+    const paginationObj = prismaUtils.getPaginationObject(page, limit);
+    filterObj = startDate ? {
+        ...filterObj, ...prismaUtils.getDateRangeObject(startDate, endDate)
```
Comment (satya-patnaik) [inline src/services/dsm/teamRequests.services.js:43]: Please create a seperate utils function and resuse the logic for assigning filterObj.

File: src/services/dsm/teamRequests.services.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            createdAt,
+            ...(taggedIndividuals && { taggedIndividuals }),
+        },
+        ...selectOnlyValidTeamrequestsFields
+    });
+    return updatedRequest;
+}
+// service to delete team request by team request id
+const deleteTeamRequest = async (requestId) => {
+    const deleteRequest = await prisma.Request.delete(
```
Comment (satya-patnaik) [inline src/services/dsm/teamRequests.services.js:93]: Please add try/catch block for error handling

File: src/services/dsm/teamRequests.services.js (hunk 1 above)
Comment (karthik01455) [reply src/services/dsm/teamRequests.services.js]: fixed

File: src/services/dsm/teamRequests.services.js (hunk 6 above)
Comment (karthik01455) [reply src/services/dsm/teamRequests.services.js:92]: will add it

File: src/services/dsm/teamRequests.services.js (hunk 5 above)
Comment (karthik01455) [reply src/services/dsm/teamRequests.services.js:42]: will add it

File: src/middlewares/index.js [hunk 7]
```diff
@@ -26,6 +25,9 @@ function errorHandlingMiddleware(err, req, res, next) {
   case Prisma.PrismaClientKnownRequestError: {
     // seperatly handling the internal db or query errors
     // thrown prisma  ("2***" error codes)
+    if(err.code === 'P2025') {
```
[the remark this answers is not in the document]
Comment (karthik01455) [reply src/middlewares/index.js:28]: will change it

File: src/utils/paramsHandling.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+const parseIntIdParam = (req, res, next) => {
+    req.params.id = parseInt(req.params.id, 10);
+    next();
+  };
+module.exports = { parseIntIdParam };
```
[the remark this answers is not in the document]
Comment (karthik01455) [reply src/utils/paramsHandling.js:5]: fixed

File: src/routes/api/dsm/teamRequests.routes.js (hunk 4 above)
Comment (karthik01455) [reply src/routes/api/dsm/teamRequests.routes.js]: fixed

File: src/middlewares/index.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (karthik01455) [reply src/middlewares/index.js]: fixed

File: src/services/dsm/teamRequests.services.js (hunk 5 above)
Comment (karthik01455) [reply src/services/dsm/teamRequests.services.js]: fixed

File: src/services/dsm/teamRequests.services.js (hunk 6 above)
Comment (karthik01455) [reply src/services/dsm/teamRequests.services.js]: added

**Pull request closed without merging**